### PR TITLE
Port's RCL from /vg/

### PR DESCRIPTION
**Pull request by davipatury**
:cl: davipatury
rscadd: Ported Rapid Cable Layer from /vg/.
/:cl:

### Pictures:
RCL

### CODE REVIEW

**Comment by SamHPurp** [comment]
Ezmode engineering. 👎

**Comment by tigercat2000** [comment]
@SamHPurp I am beginning to think you intentionally oppose all QoL features...

**Comment by Anasari** [comment]
This seems to be a great idea. If we port in engineering point, making it part of a reward would be nice too.

**Comment by SamHPurp** [comment]
@tigercat2000 false, I supported the Magistrate getting white gloves.

**Comment by fludd12** [comment]
@Anasari GBPoints definitely seem like they'd fit paradise more than they'd fit TG

**Comment by Crazylemon64** [comment]
my goodness was this needed for a long time

**Comment by Fox-McCloud** [comment]
The sprite is terribad, but the feature is super good.

**Comment by davidchan13** [comment]
@SamHPurp and opposed everything else in that PR. Gloves aren't QoL, not having to jump through hoops to do you job is.

**Comment by SamHPurp** [comment]
@davidchan13 ultimately there isn't enough work for Engineers to do as there is, effectively removing a key part of their job seems pointless to me, and is a reduction in quality of life, because now I have less things to do.

**Comment by uraniummeltdown** [comment]
@taukausanake hacked autolathe

File: code/game/objects/items/weapons/RCL.dm [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	icon_state = "rcl-0"
+	item_state = "rcl-0"
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	opacity = 0
+	siemens_coefficient = 1 //Not quite as conductive as working with cables themselves
+	force = 5.0 //Plastic is soft
+	throwforce = 5.0
+	throw_speed = 1
+	throw_range = 10
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/RCL.dm]: 7

File: code/game/objects/items/weapons/RCL.dm [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	name = "rapid cable layer (RCL)"
+	desc = "A device used to rapidly deploy cables. It has screws on the side which can be removed to slide off the cables."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "rcl-0"
+	item_state = "rcl-0"
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	opacity = 0
+	siemens_coefficient = 1 //Not quite as conductive as working with cables themselves
+	force = 5.0 //Plastic is soft
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/RCL.dm]: no need for `.0`

File: code/game/objects/items/weapons/RCL.dm [hunk 3]
```diff
@@ -0,0 +1,154 @@
+/obj/item/weapon/twohanded/rcl
+	name = "rapid cable layer (RCL)"
+	desc = "A device used to rapidly deploy cables. It has screws on the side which can be removed to slide off the cables."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "rcl-0"
+	item_state = "rcl-0"
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/RCL.dm]: isn't this already defined by parent?

File: code/game/objects/items/weapons/RCL.dm [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	if(loaded)
+		to_chat(user, "<span class='info'>It contains [loaded.amount]/[max_amount] cables.</span>")
+
+/obj/item/weapon/twohanded/rcl/Destroy()
+	if(loaded)
+		qdel(loaded)
+		loaded = null
+	last = null
+	active = 0
+	..()
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/RCL.dm]: `return ..()`

File: code/game/objects/items/weapons/RCL.dm [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+			item_state = "rcl"
+		else
+			icon_state = "rcl-0"
+			item_state = "rcl-0"
+
+/obj/item/weapon/twohanded/rcl/proc/is_empty(mob/user, loud = 1)
+	update_icon()
+	if(!loaded || !loaded.amount)
+		if(loud)
+			to_chat(user, "<span class='notice'>The last of the cables unreel from \the [src].</span>")
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/RCL.dm]: `from [src]`

File: code/game/objects/items/weapons/RCL.dm [hunk 6]
```diff
@@ -0,0 +1,154 @@
+/obj/item/weapon/twohanded/rcl
+	name = "rapid cable layer (RCL)"
+	desc = "A device used to rapidly deploy cables. It has screws on the side which can be removed to slide off the cables."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "rcl-0"
+	item_state = "rcl-0"
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	opacity = 0
+	siemens_coefficient = 1 //Not quite as conductive as working with cables themselves
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/RCL.dm]: I think this is the default value, too.

**Comment by davidchan13** [comment]
@SamHPurp While I do agree that engineers don't have enough work. Making what work they have to do tedious and annoying when alternative options exist.

Maybe this will lead to people giving more stuff for engineering to do, maybe it won't. Increasing their ability to reliably set up new areas of the station or repair damaged ones isn't a bad thing in my book. And just imagine if a clever coder figured out a way to make det cord or fuses that can use this mechanic?

Sprite needs work though.

**Pull request merged**